File: laramie-tesoreria-webapi/src/domain/entities/registro-contable-lote.ts
export default class RegistroContableLote {

    id: number;
	numeroLote: string;
	fechaLote: Date;
	casos: number;
	importeTotal: number;
	idUsuarioProceso: number;
	fechaProceso: Date;
	fechaConfirmacion: Date;
	pathArchivoRegistroContable: string;

	constructor(
        id: number = 0,
		numeroLote: string = "",
		fechaLote: Date = null,
		casos: number = 0,
		importeTotal: number = 0,
		idUsuarioProceso: number = 0,
		fechaProceso: Date = null,
		fechaConfirmacion: Date = null,
		pathArchivoRegistroContable: string = ""
	)
	{
        this.id = id;
		this.numeroLote = numeroLote;
		this.fechaLote = fechaLote;
		this.casos = casos;
		this.importeTotal = importeTotal;
		this.idUsuarioProceso = idUsuarioProceso;
		this.fechaProceso = fechaProceso;
		this.fechaConfirmacion = fechaConfirmacion;
		this.pathArchivoRegistroContable = pathArchivoRegistroContable;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.numeroLote = row.numeroLote ?? "";
		this.fechaLote = row.fechaLote ?? null;
		this.casos = row.casos ?? 0;
		this.importeTotal = row.importeTotal ?? 0;
		this.idUsuarioProceso = row.idUsuarioProceso ?? 0;
		this.fechaProceso = row.fechaProceso ?? null;
		this.fechaConfirmacion = row.fechaConfirmacion ?? null;
		this.pathArchivoRegistroContable = row.pathArchivoRegistroContable ?? "";
	}

}

File: modules/tesoreria/schemas/registro_contable_lote.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, datetime
from decimal import Decimal


class RegistroContableLoteBase(BaseModel):
    numero_lote: int
    fecha_lote: date


class RegistroContableLoteCreate(RegistroContableLoteBase):
    pass


class RegistroContableLoteUpdate(BaseModel):
    numero_lote: Optional[int] = None
    fecha_lote: Optional[date] = None
    casos: Optional[int] = None
    importe_total: Optional[Decimal] = None
    id_usuario_proceso: Optional[int] = None
    fecha_proceso: Optional[datetime] = None
    fecha_confirmacion: Optional[datetime] = None
    path_archivo_registro_contable: Optional[str] = None


class RegistroContableLoteResponse(BaseModel):
    id: int
    numero_lote: int
    fecha_lote: date
    casos: int = 0
    importe_total: Decimal = Decimal("0")
    id_usuario_proceso: Optional[int] = None
    fecha_proceso: Optional[datetime] = None
    fecha_confirmacion: Optional[datetime] = None
    path_archivo_registro_contable: Optional[str] = None

    class Config:
        from_attributes = True
